### PR TITLE
Improving race-swap framework and adding Smash and Grab Z/P

### PR DESCRIPTION
**Pull request by EnvyDragon**
Fixes/polishes some of the code in the initial faction-swap framework (ex. creep tumor deletion, larva count) and implements the framework into Smash and Grab. Changes to that map will be no-op unless the client explicitly requests that it be initialized as a particular variant.

Smash and Grab (Z):
- Starting units are replaced with 6 zerglings and 1 queen
- Marauder reinforcements are replaced with 4 roaches (3 on brutal) and 1 ravager, arriving by drop pod. The voice lines are also changed
- All base structures are replaced

Smash and Grab (P):
- Starting units are replaced with 2 zealots
- Marauder reinforcements are replaced with 3 warp-in dragoons. The voice lines are also changed
- All base structures are replaced

Note that these mission variants will be pretty hard to access without the accompanying generator change
(if you want to try it out, you can load it in the editor and send a valid ?SetOptions string with either 2 (for zerg) or 3 (for protoss) appended to the end before sending ?LoadFinished)

### CODE REVIEW

File: Mods/ArchipelagoTriggers.SC2Mod/Base.SC2Data/LibABFE498B.galaxy [hunk 1]
```diff
@@ -82,6 +83,23 @@ void libABFE498B_gf_AP_Triggers_MapConfig_setSoAControllingPlayers (playergroup
...
+    else if ((lp_race == "2")) {
+        libABFE498B_gv_aP_Triggers_Option_overridePlayerRace = "Zerg";
+        lib5BD4895D_gf_AP_Core_MapConfig_setPlayerFaction(1, lib5BD4895D_gv_aP_Core_Faction_KERRIGAN_SWARM);
+    }
+    else if ((lp_race == "3")) {
+        libABFE498B_gv_aP_Triggers_Option_overridePlayerRace = "Prot";
+        lib5BD4895D_gf_AP_Core_MapConfig_setPlayerFaction(1, lib5BD4895D_gv_aP_Core_Faction_DAELAAM);
+    }
+}
+
```
[the remark this answers is not in the document]
Comment (EnvyDragon) [reply Mods/ArchipelagoTriggers.SC2Mod/Base.SC2Data/LibABFE498B.galaxy:102]: Overriding player race _is_ done via bot command (this trigger is called in the SetOptions parser), but you're right that I shouldn't have put "MapConfig" in the trigger name or even changed faction variables here. I'll clean that up.

File: Mods/ArchipelagoTriggers.SC2Mod/Base.SC2Data/LibABFE498B.galaxy [hunk 2]
```diff
@@ -7004,6 +7024,7 @@ bool libABFE498B_gt_AP_Triggers_setOptions_Func (bool testConds, bool runActions
     lv_ncoOnlyRun = StringWord(EventChatMessage(false), 13);
     lv_storyLevels = StringWord(EventChatMessage(false), 14);
     lv_enableMorphling = StringWord(EventChatMessage(false), 15);
+    lv_overrideRace = StringWord(EventChatMessage(false), 16);
```
Comment (Ziktofel) [inline Mods/ArchipelagoTriggers.SC2Mod/Base.SC2Data/LibABFE498B.galaxy:7027]: Need to reserve that on Client side (make a PR for that, If the value is `0` sent from the client, do not override)

Possibly make it dummy, let the client send always `0` (with comment Player race override)

Comment (EnvyDragon) [reply Mods/ArchipelagoTriggers.SC2Mod/Base.SC2Data/LibABFE498B.galaxy:7027]: I agree, yeah. That's what I've been doing in my testing, it's just that the actual logic of adding map variations has turned out to be a lot more complicated than I'd expected (and I've been too busy to work on it for a little while). I've converted this PR to a draft so it's clearer that it's not ready to merge yet.

Comment (EnvyDragon) [reply Mods/ArchipelagoTriggers.SC2Mod/Base.SC2Data/LibABFE498B.galaxy:7027]: screw it, I'm just gonna make a client PR to reserve the 16th word of the string like you suggested, the generator logic isn't going to be ready for a while and I want to put this through before too many conflicts arise

Comment (EnvyDragon) [reply Mods/ArchipelagoTriggers.SC2Mod/Base.SC2Data/LibABFE498B.galaxy:7027]: https://github.com/Ziktofel/Archipelago/pull/214

File: Mods/ArchipelagoTriggers.SC2Mod/Base.SC2Data/LibABFE498B.galaxy (hunk 1 above)
[the remark this answers is not in the document]
Comment (EnvyDragon) [reply Mods/ArchipelagoTriggers.SC2Mod/Base.SC2Data/LibABFE498B.galaxy]: Update: changed trigger name and scope, all it does now is set an override variable. The map itself is now going to set the MapConfig player faction variable based on what's in that override (see smash+grab triggers for an example)

**Pull request merged**